charts/datadog/templates/cluster-agent-deployment.yaml

```diff
@@ -266,6 +266,10 @@ spec:
           - name: DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_IAST_ENABLED
             value: "true"
           {{- end }}
+          {{- if .Values.datadog.profiling.enabled }}
+          - name: DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_PROFILING_ENABLED
+            value: {{ .Values.datadog.profiling.enabled | quote }}
+          {{- end }}
```

Comment:
After the reading the doc you created. I don't think this `if` does what you want it to do about `nil` values. cf [helm chart docs](https://helm.sh/docs/chart_template_guide/control_structures/#ifelse)

Reply:
I _think_ it does what I want (omits the env var on null), but it doesn't do what I want for "false" (which would be emitting explicit "false" value, but it also omits the env var) so I need to change that. Right now it's:
```
% helm template ./datadog --set datadog.profiling.enabled=null | grep PROFILING -A 1
% helm template ./datadog --set datadog.profiling.enabled=false | grep PROFILING -A 1
% helm template ./datadog --set datadog.profiling.enabled=auto | grep PROFILING -A 1
          - name: DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_PROFILING_ENABLED
            value: "auto"
% helm template ./datadog --set datadog.profiling.enabled=true | grep PROFILING -A 1
          - name: DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_PROFILING_ENABLED
            value: "true"
```
but with a change to the predicate to say `{{- if not (eq .Values.datadog.profiling.enabled nil) }}` it's doing what I'd expect it to. If there's maybe some other aspect that I'm not seeing, please let me know.